Compile fixes for ITKv5 with legacy disabled

dzenanz opened the pull request.

dzenanz committed:
COMP: add ITK_ prefix to testing macros

dzenanz committed:
COMP: Do not include itkExceptionObject.h directly

dzenanz committed:
COMP: fully qualify enumeration names

dzenanz committed:
COMP: rename GetNumberOfThreads into GetNumberOfWorkUnits

dzenanz committed:
COMP: fix signed/unsigned comparison warning

jhlegarreta commented on examples/computeKrcahBoneEnhancement.cxx:
```diff
@@ -64,7 +64,7 @@ int main(int argc, char * argv[])
   double thisSigma;
   itk::Array< double > sigmaArray;
   sigmaArray.SetSize(numberOfSigma);
-  for (unsigned int i = 0; i < numberOfSigma; ++i) {
```
May be `numberOfSigma` should be `unsigned`?

jhlegarreta reviewed (commented):
An in-line comment.

The rest of the changes seem OK to me.

dzenanz committed:
COMP: fix warnings -Winconsistent-missing-override

e.g. warning: 'PrintSelf' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]

dzenanz committed:
COMP: fix GCC -Wall warnings

jhlegarreta commented:
Sorry, although this has been merged, wondering why [this line](https://github.com/InsightSoftwareConsortium/ITKBoneEnhancement/blob/00f62b474a74079d04397bad881b080743f41b9d/examples/computeKrcahBoneEnhancement.cxx#L67) does not throw an `unsigned long` vs. `unsigned int` comparison warning now.

dzenanz commented:
Because I changed the declaration of `numberOfSigma` from `int` to `ulong`:
https://github.com/InsightSoftwareConsortium/ITKBoneEnhancement/blob/00f62b474a74079d04397bad881b080743f41b9d/examples/computeKrcahBoneEnhancement.cxx#L63
in response to your earlier comment.

jhlegarreta commented:
Yes, I realized @dzenanz , but by the same token you first proposed to change the `for` loop counter type, I'd expect now an `unsigned long` vs. `unsigned int` warning.

dzenanz commented:
When two unsigned numbers are compared, the smaller one is extended to the size of the larger one (I think), so no problems and no warning.

jhlegarreta commented:
OK. Thanks @dzenanz .
